CMSgov/cms-mdct-seds
# Local login cog

**Purpose**

The purpose of this PR is to create a remote login feature that allows developers to login locally via cognito to their OWN branch if need be, you can still sign-in with "Alice" locally. This creates and  enables the ability to test fully on a aws developer branch.

**Linked Issues to Close**
Addresses JIRA ticket: https://qmacbis.atlassian.net/browse/OY2-6500

**Approach**
The approach for this was so create an environment that allows each developer to be able to login remotely, this isnt a new feature, it just wasn't finished hooked up

**Test(login in remotely - Workflow )**

1. branch off master and push your branch up
2. Verify that you userpool was created correctly in AWS
3. retrieve your AWS keys (using ctkey) and run the exports (i will provided more documentation on how to setup ctkeys)
4. run the configureLocal.sh script which is located in the root of ui-src using this command "sh configureLocal.sh branchName"
5. This script will run and retrieve all the cognito env required for your specific branch.
6. Then navigate to localhost:3000/signup
7. Your only signup once, and after that you should be able to login and out as far has your branch exists on only locally but on the cloud.

**Pull Request Creator Checklist**

- [ ]  I updated the architecture diagram if applicable

**Pull Request Reviewer/Assignee Checklist**
